pkg/render/apiserver.go

```diff
@@ -197,12 +196,18 @@ func (c *apiServerComponent) Objects() ([]client.Object, []client.Object) {
 		globalObjects, objsToDelete = populateLists(globalObjects, objsToDelete, c.apiServerPodSecurityPolicy)
 	}
 
-	if c.cfg.ManagementCluster != nil && c.cfg.MultiTenant {
-		// Multi-tenant management cluster API servers need access to per-tenant CA secrets in order to sign
-		// per-tenant guardian certificates when creating ManagedClusters.
-		globalObjects = append(globalObjects, c.secretsClusterRole()...)
+	if c.cfg.ManagementCluster != nil {
+		if c.cfg.MultiTenant {
+			// Multi-tenant management cluster API servers need access to per-tenant CA secrets in order to sign
+			// per-tenant guardian certificates when creating ManagedClusters.
+			globalObjects = append(globalObjects, c.secretsClusterRole()...)
+		} else {
+			globalObjects = append(globalObjects, c.secretsRole()...)
+		}
 	} else {
-		objsToDelete = append(objsToDelete, c.secretsClusterRole()...)
+		if c.cfg.MultiTenant {
```

Comment:
I don't think this branch is valid - multi-tenant clusters _must_ be management clusters, so you should never hit this else if case

Reply:
Not sure I follow. If we are running multi-tenant, we install a role instead of a cluster role. Are you saying that the role is not needed ?

Reply:
No, I mean that this branch is logically only ever hit if `c.cfg.ManagementCluster == nil && c.cfg.MultiTenant` - which is not a valid combination. There's no such thing as a "multi-tenant standalone cluster"

Reply:
In other words, I think this should be this:

```go
} else {
  // If we're not a management cluster, the API server doesn't need permissions to access secrets.
  objsToDelete = append(objsToDelete, c.secretsClusterRole()...)
  objsToDelete = append(objsToDelete, c.secretsRole()...)
}
```